### Ingest

In [1]:
# Imports
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split

# Display all the columns no matter how many
pd.pandas.set_option("display.max_columns", None)

In [ ]:
from wrangle import ingest_and_split_train_val

In [3]:
train, val =  ingest_and_split_train_val()

wrangle

In [ ]:
from wrangle import wrangle

In [5]:
train.shape, val.shape

((1168, 80), (292, 80))

In [6]:
train = wrangle(train)
val = wrangle(val)

In [7]:
train.shape, val.shape

((1168, 57), (292, 57))

In [8]:
# Split into features and labels
X_train = train.drop(columns=["SalePrice"])
X_val = val.drop(columns=["SalePrice"])
y_train = train["SalePrice"]
y_val = val["SalePrice"]

In [9]:
X_train.shape, X_val.shape, y_train.shape, y_val.shape

((1168, 56), (292, 56), (1168,), (292,))

### Baseline

The baseline is basically a naive model. It will predict only the mean of the train dataset

In [11]:
# y_train

In [20]:
y_mean = y_train.mean()
y_train_pred_baseline = [y_mean] * len(y_train)
y_train_pred_baseline[:5]

[181100.5522260274,
 181100.5522260274,
 181100.5522260274,
 181100.5522260274,
 181100.5522260274]

In [16]:
from sklearn.metrics import mean_squared_error

In [21]:
train_rmse = mean_squared_error(y_train, y_train_pred_baseline, squared=False)
train_rmse

80963.53540864638

In [22]:
y_val_pred_baseline = [y_mean] * len(y_val)

In [24]:
val_rmse = mean_squared_error(y_val, y_val_pred_baseline, squared=False)
val_rmse

72895.3188260644

A naive model gave us a RMSE of 80,693 on the training set
It also gave us a rmse of 72,895 on the validation set